### ISSUE TITLE
FP S3928: should not raise an issue in set/init, when nameof(property_name) is used

### ISSUE THREAD

**Issue by dmitry-golovinov-sonarsource** [SonarSource/sonar-dotnet#4679]
### Description

Rule should not raise an issue in set/init parts of property, when `nameof(property name)` is used.
Using `nameof(value)` here makes no sense, since it produces the exception text that is too generic.

`Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'value')`
### Repro steps
```
    public class InitOnlyProperties
    {
        private readonly string firstName;
	public string FirstName
        {
            get => firstName;
            init => firstName = (value ?? throw new ArgumentNullException(nameof(FirstName)));
        }
    }
```

### Expected behavior
Issue should not be raised.

### Actual behavior
Issue is raised.

### Related information

* C#/VB.NET Plugins version 8.26
